### PR TITLE
nmcli: Treat order as significant when comparing address lists

### PR DESCRIPTION
**Pull request by jikamens**
Don't sort the old and new values for ipv4.addresses and ipv6.addresses before comparing them, because order matters in these parameters: the first address specified is the default source address for outbound connections.

##### SUMMARY

The `nmcli` module (before this change) treated the order of IPv4 or IPv6 addresses for an interface as insignificant, i.e., it would not consider it a change for the order of these addresses to change. This is not correct, since the first address for an interface is the default address for outbound connections, so order matters. The change in this PR continues to treat order as insignificant for other list settings, but treats order as significant for ipv4.addresses and ipv6.addresses.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
nmcli

### CODE REVIEW

File: changelogs/fragments/6048-nmcli-addres-order.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - nmcli module - order is significant for lists of addresses (https://github.com/ansible-collections/community.general/pull/6048).
```
Comment (felixfontein) [inline changelogs/fragments/6048-nmcli-addres-order.yml:2]:
```suggestion
  - nmcli - order is significant for lists of addresses (https://github.com/ansible-collections/community.general/pull/6048).
```

**Comment by felixfontein** [comment]
CC @ThomasGebert

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this this weekend.

**Review by ThomasGebert — APPROVED** [review]
lgtm

**Comment by felixfontein** [comment]
@jikamens thanks again for your contribution!
@ThomasGebert thanks again for reviewing!

**Comment by jikamens** [comment]
FYI: https://blog.kamens.us/2023/02/25/the-joy-of-contributing-to-a-well-run-open-source-project/

**Pull request merged**